src/components/DocumentViewer.tsx

```diff
@@ -1,22 +1,36 @@
-// @ts-ignore
-import React, { useRef, useEffect } from 'react'
-import useInstances from '../context'
+import React from 'react';
+import useInstance from '../context';
+
+const disabledElements = [
+  'header',
+  'toolsHeader',
+  'viewControlsButton',
+  'leftPanelButton',
+  'pageNavOverlay',
+  'searchButton',
+  'menuButton',
+  'textPopup',
+  'contextMenuPopup',
+];
 
 export type TProps = {
-  docUrl: string
-  UID: string
+  zoom: number
+  // will add more initial configs
 }
 
-function DocumentViewer({ docUrl, UID }: TProps): JSX.Element {
-  const viewer = useRef<HTMLDivElement>(null)
+const DocumentViewer = React.forwardRef<HTMLDivElement, TProps>((props, ref) => {
 
-  const { addInstance } = useInstances()
+  const { instance } = useInstance();
+  const docViewer = instance?.Core.documentViewer;
 
-  useEffect(() => {
-    addInstance(docUrl, UID, viewer.current)
-  }, [])
+  if (instance) {
+    instance.UI.disableElements(disabledElements);
+    docViewer?.addEventListener('documentLoaded', function() {
+      instance.UI.setZoomLevel(props.zoom);
+    });
+  }
```

Comment:
This needs to be in a useEffect or else it will get executed every render

Reply:
Actually this code should not even exist

Reply:
I had misunderstanding on the expectation of this component. Removed all these code.